From 97d0bd96c536349f21f6b6bc1f148f7ba58f2990 Mon Sep 17 00:00:00 2001
From: Ben Skeggs <bskeggs@redhat.com>
Date: Mon, 11 Oct 2010 10:35:43 +1000
Subject: [PATCH] nva3-noaccel-info

Add info to logs as to likely reason why acceleration isn't available
on NVA3+ chipsets.
---
 src/nv_dma.c |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/src/nv_dma.c b/src/nv_dma.c
index 49ed40a..2d837ea 100644
--- a/src/nv_dma.c
+++ b/src/nv_dma.c
@@ -56,6 +56,21 @@ NVInitDma(ScrnInfoPtr pScrn)
 	if (ret) {
 		xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
 			   "Error creating GPU channel: %d\n", ret);
+		switch (pNv->dev->chipset) {
+		case 0xa3:
+		case 0xa5:
+		case 0xa8:
+		case 0xaf:
+			xf86DrvMsg(pScrn->scrnIndex, X_INFO,
+				   "Acceleration disabled by default due to "
+				   "unresolved GPU hangs on your chipset.\n");
+			xf86DrvMsg(pScrn->scrnIndex, X_INFO,
+				   "See https://bugzilla.redhat.com/"
+				   "show_bug.cgi?id=625187 for more "
+				   "information\n");
+			break;
+		}
+
 		return FALSE;
 	}
 	pNv->chan->user_private = pScrn;
-- 
1.7.4.1

